configure

```diff
@@ -2275,35 +2272,20 @@ if test "$_release_build" = yes; then
 	append_var DEFINES "-DRELEASE_BUILD"
 fi
 
-# If possible, we want to use -Wglobal-constructors
-# However, not all compilers support that, so check whether the active one does.
-echocheck "whether C++ compiler accepts -Wglobal-constructors"
-cat > $TMPC << EOF
-int main() { return 0; }
-EOF
-cc_check -Wglobal-constructors -Werror && _global_constructors=yes
-
-if test "$_global_constructors" = yes; then
-	append_var CXXFLAGS "-Wglobal-constructors"
-fi
-echo $_global_constructors
-
-# If the compiler supports the -Wundefined-var-template flag, silence that warning.
-# We get this warning a lot with regard to the Singleton class as we explicitly
-# instantiate each specialisation. An alternate way to deal with it would be to
-# change the way we instantiate the singleton classes as done in PR #967.
-# Note: we check the -Wundefined-var-template as gcc does not error out on unknown
-# -Wno-xxx flags.
-echocheck "whether C++ compiler accepts -Wno-undefined-var-template"
-cat > $TMPC << EOF
-int main() { return 0; }
+set_flag_if_supported() {
+	echocheck "whether C++ compiler accepts $1"
+	cat > $TMPC << EOF
+	int main() { return 0; }
 EOF
-cc_check -Wundefined-var-template -Werror && _no_undefined_var_template=yes
+	if cc_check $1; then
+		append_var CXXFLAGS "$1"
+		echo yes
+	else
+		echo no
+	fi
+}
 
-if test "$_no_undefined_var_template" = yes; then
-	append_var CXXFLAGS "-Wno-undefined-var-template"
-fi
-echo $_no_undefined_var_template
```

Comment:
This whole check shouldn't be removed.
The PR got reverted in 940b2a20f1cd490afb6e541a3cd26f0d3bdd1687.

Reply:
Which compilers have this warning? I don't have it with MinGW/GCC12.

Reply:
Reverted.

Reply:
Maybe clang?

Reply:
Yes. I see it here: https://github.com/orgads/scummvm/runs/6715783766?check_suite_focus=true#step:8:474

Reverted anyway, thank you.